src/mixer/softmixer.rs

```diff
@@ -0,0 +1,56 @@
+use super::Mixer;
+use super::StreamEditor;
+use std::borrow::Cow;
+use std::sync::{Arc, RwLock};
+
+pub struct SoftMixer {
+  volume: Arc<RwLock<u16>>
```

Comment:
Make that an atomic.
`AtomicU16` is unstable, but you can use `AtomicUsize` and cast values back and forth